generate_workspace/src/main/java/com/google/devtools/build/workspace/maven/Resolver.java

```diff
@@ -163,18 +163,34 @@ void traverseDeps(Model model, Set<String> exclusions, Rule parent) {
       }
     }
     for (Dependency dependency : model.getDependencies()) {
-      addDependency(dependency, model, exclusions, parent);
+      addDependency(dependency, model, scopes, exclusions, parent);
     }
   }
 
   private void addDependency(
       Dependency dependency,
       Model model,
+      Set<String> topLevelScopes,
       Set<String> exclusions,
       @Nullable Rule parent) {
+    String scope = dependency.getScope();
     // DependencyManagement dependencies don't have scope.
-    if (dependency.getScope() != null && !dependency.getScope().equals(COMPILE_SCOPE)) {
-      return;
+    if (scope != null) {
+      if (parent == null) {
+        // Top-level scopes get pulled in based on the user-provided scopes.
+        if (!topLevelScopes.contains(scope)) {
+          return;
+        }
+      } else {
```

Comment:
Can you expand on what you mean by "relabel the transitive dependencies"?

For dependencies that are not scoped as compile or runtime, what would they be relabeled as?

Reply:
The issue is that transitive dependencies are handled identically regardless of the scope on which the root project depends on them.

For example, if the root project depends on `junit` in the `test` scope, all of `junit`s `compile` dependencies will be treated as a `compile` dependency of the root project, instead of as a `test` dependency.

This doesn't matter much if we're not generating a `java_test` target.

Reply:
I see. So you are saying, the scope of the parent should propagate to all of its children. That makes sense. 

Minor nit, but can you rephrase the TODO so that it says something like:
```
// TODO (bazel-devel): Relabel the scope of transitive dependencies so that it matches the scope of it's parent, 
// e.g. if root project depends on junit with test scope, and
// all of junit's dependencies are labeled as compile, they should be retained as test
```

Reply:
It should propagate as in the maven rules, linked in the comment. These match except for compile->runtime dependencies, which is handled as runtime instead of compile.

Are you asking just to update the comment, or to update the logic fully? I can't do the latter without adding a scope field to `Rule`, and I wasn't sure if that was desirable.

Reply:
Just the change to the comments.

Reply:
Done.